modules/storage/storage-account/main.bicep

```diff
@@ -19,6 +19,100 @@ param isZoneRedundant bool = false
 @description('Storage Account Type. Use Zonal Redundant Storage when able.')
 param storageAccountType string = isZoneRedundant ? 'Standard_ZRS' : 'Standard_LRS'
 
+@description('It will be moved to the cool tier after the given amount of days.')
+param daysAfterLastModification int = 30
+
+@description('Specifies the type of blob to manage the lifecycle policy.')
+param blobType string = 'blockBlob'
+
+@description('Indicates whether change feed event logging is enabled for the Blob service.')
+param changeFeedEnabled bool = false
+
+@description('Versioning is enabled if set to true. ')
+param versioningEnabled bool = false
+
+@description('Allows https traffic only to storage service if sets to true.')
+param supportHttpsTrafficOnly bool = true
+
+@description('Allow or disallow public access to all blobs or containers in the storage account.')
+param allowBlobPublicAccess bool = false
+
+@description('Replication of objects between AAD tenants is allowed or not. For this property, the default interpretation is true.')
+param allowCrossTenantReplication bool = false
+
+@description('Allow or disallow public network access to Storage Account. Value is optional but if passed in, must be Enabled or Disabled.')
+param publicNetworkAccess string = 'Disabled'
+
+@description('Set the minimum TLS version to be permitted on requests to storage. The default interpretation is TLS 1.0 for this property.')
+param minimumTlsVersion string = 'TLS1_2'
```

Comment:
should be able to use an @Allowed here for the allowed versions.

Reply:
@allowed added here.